src/main/java/org/jenkinsci/plugins/github_branch_source/ForkPullRequestDiscoveryTrait.java

```diff
@@ -316,6 +318,94 @@ public boolean isApplicableToOrigin(@NonNull Class<? extends SCMHeadOrigin> orig
     }
 
     /**
+     * An {@link SCMHeadAuthority} that trusts contributors to the repository.
+     */
+    public static class TrustPermission
+            extends SCMHeadAuthority<GitHubSCMSourceRequest, PullRequestSCMHead, PullRequestSCMRevision> {
+        @NonNull
+        private final GHPermissionType permission;
+
+        /**
+         * Constructor.
+         */
+        @DataBoundConstructor
+        public TrustPermission(@NonNull String permission) {
+            GHPermissionType permissionType = GHPermissionType.ADMIN;
```

Comment:
Why is this even configurable? The only useful value is `WRITE`. If you can push to the repository, you could change the `Jenkinsfile`, thus you are trusted.